### PR TITLE
MegaMenu: Fix broken hamburger toggle

### PR DESCRIPTION
**Pull request by kitsuned**
**What this PR does / why we need it**:

This PR changes the behavior of `<NavBarMenu />`'s onClose prop: now it fires after the CSS transition is completed.

**Which issue(s) this PR fixes**:

See https://github.com/grafana/grafana/pull/51616#pullrequestreview-1028505298.
related: https://github.com/grafana/grafana/issues/50791

**Special notes for your reviewer**:

Now `<MegaMenu />` wouldn't be unmounted, if this is unacceptable – let's discuss it!

### CODE REVIEW

**Comment by torkelo** [comment]
@kitsuned just curious how did you find this bug? are you testing the new topnav feature (work in progress chance) ?

**Comment by kitsuned** [comment]
> @kitsuned just curious how did you find this bug? are you testing the new topnav feature (work in progress chance) ?

I've been looking at v9.1.0 milestone and got interested about #50791.
So I just decided to try out the new navigation & found this bug!

**Comment by torkelo** [comment]
@kitsuned that is great! really excited about the new nav & page layouts, it won't make it into 9.1, a lot of pages and plugins (apps) that needs to be updated but hopefully in 9.2 it should be ready

**Comment by kitsuned** [comment]
@torkelo I think I can contribute a little bit on the new navigation. 🔥

File: public/app/core/components/MegaMenu/NavBarMenu.tsx [hunk 1]
```diff
@@ -27,56 +28,74 @@ export function NavBarMenu({ activeItem, navItems, searchBarHidden, onClose }: P
...
             />
+          </div>
+          <NavBarToggle
+            className={styles.menuCollapseIcon}
+            isExpanded={true}
+            onClick={() => {
+              reportInteraction('grafana_navigation_collapsed');
+              onMenuClose();
+            }}
+          />
```
Comment (ashharrison90) [inline public/app/core/components/MegaMenu/NavBarMenu.tsx:82]: don't think we can move this outside of the `FocusScope` as there's then no tabbable element to focus which will close the menu 🤔

**Comment by kitsuned** [comment]
@ashharrison90 okay, I moved toggles to the focus scope. 👌

image

**Review by ashharrison90 — APPROVED** [review]
thanks for this, looks great behaviour wise 🙌

**Pull request merged**